src/test/regress/expected/multi_router_planner.out

```diff
@@ -2151,7 +2151,7 @@ GRANT INSERT ON ALL TABLES IN SCHEMA public TO router_user;
 BEGIN;
 INSERT INTO failure_test VALUES (1, 1);
 WARNING:  connection error: localhost:57638
-DETAIL:  no connection to the server
+DETAIL:  FATAL:  role "router_user" does not exist
```

Comment:
Why did this message change?

Reply:
(I like that it did, just wondering what changed about the code path)

Reply:
It looks like, we used to try sending the query even if the connection was not okay. Then, this was overwriting the error message in the `conn->errorMessage.data;` and we were getting `no connection to the server` error.

Here is the call stack
```
SendRemoteCommand() at remote_commands.c:462 0x10457ade0	
StartRemoteTransactionBegin() at remote_transaction.c:102 0x1045cba02	
RemoteTransactionsBeginIfNecessary() at remote_transaction.c:601 0x1045cc9b3	
GetModifyConnections() at multi_router_executor.c:926 0x104581ddd	
ExecuteSingleModifyTask() at multi_router_executor.c:731 0x1045806d6	
RouterSequentialModifyExecScan() at multi_router_executor.c:484 0x10458053d	
ExecCustomScan() at nodeCustom.c:118 0x10384b3e8	
ExecProcNodeFirst() at execProcnode.c:430 0x103835842	
ExecProcNode() at executor.h:250 0x1038325b2	
ExecutePlan() at execMain.c:1,721 0x10382e346	
standard_ExecutorRun() at execMain.c:363 0x10382e21b	
ExecutorRun() at execMain.c:306 0x10382dfd7	
ProcessQuery() at pquery.c:161 0x103a3ad1e	
PortalRunMulti() at pquery.c:1,286 0x103a39d88	
PortalRun() at pquery.c:799 0x103a39464	
exec_simple_query() at postgres.c:1,099 0x103a34cec	
PostgresMain() at postgres.c:4,090 0x103a33ef7	
BackendRun() at postmaster.c:4,357 0x10397ae5e	
BackendStartup() at postmaster.c:4,029 0x10397a1d3	
ServerLoop() at postmaster.c:1,753 0x1039791c5	
<...more frames...>	
```